astropy/modeling/powerlaws.py

```diff
@@ -129,6 +129,142 @@ def fit_deriv(x, amplitude, x_break, alpha_1, alpha_2):
         return [d_amplitude, d_x_break, d_alpha_1, d_alpha_2]
 
 
+class SmoothlyBrokenPowerLaw1D(Fittable1DModel):
+    """One dimensional smoothly broken power law model.
+
+    Parameters
+    ----------
+    amplitude : float
+        Model amplitude at the break point
+    x_break : float
+        Break point
+    alpha_1 : float
+        Power law index for x << x_break
+    alpha_2 : float
+        Power law index for x >> x_break
+    smooth : float
+        Smoothness parameter
+
+    See Also
+    --------
+    BrokenPowerLaw1D
+
+    Notes ----- Model formula (with :math:`A` for ``amplitude`` and
+    :math:`\\alpha_1` for ``alpha_1`` and :math:`\\alpha_2` for
+    ``alpha_2`` and :math:`C` for ``smooth`` and :math:`S` for the
+    sign of ``alpha_1-alpha_2``):
+
+        .. math::
+
+            f(x) = A (x / x_break) ^ {-\\alpha_1}
+                   \\left[
+                      \\frac{1 + (x / x_{break})^{|\\alpha_1-\\alpha_2| * C}}{2}
+                   \\right]}^{S / C}
+
+    """
+
+    amplitude = Parameter(default=1, bounds=[1.e-10, None])
+    x_break = Parameter(default=1, bounds=[1.e-10, None])
+    alpha_1 = Parameter(default=1, bounds=[-6, 6])
+    alpha_2 = Parameter(default=1, bounds=[-6, 6])
+    smooth = Parameter(default=1, bounds=[1.e-4, 1.e4])
+
+    @staticmethod
+    def evaluate(x, amplitude, x_break, alpha_1, alpha_2, smooth):
+        """One dimensional smoothly broken power law model function"""
+
+        if (amplitude <= 0):
+            raise ValueError("amplitude must be positive ("+str(amplitude)+")")
+        if (x_break <= 0):
+            raise ValueError("x_break value must be positive ("+str(x_break)+")")
+        if (smooth <= 0):
+            raise ValueError("smooth value must be positive ("+str(smooth)+")")
+
+        xx = x / x_break
+        logt =  np.abs(alpha_1 - alpha_2) * smooth * np.log(xx)
+        sign = np.sign(alpha_1 - alpha_2)
+        if (sign == 0):
+            sign = 1
+
+        f = xx*0.
+
+        i = np.where(logt > 30)
```

Comment:
Magic number! Where did this number come from and why does it need to be exactly 30? This comment also applies to all other magic numbers below.